### PR TITLE
Sdk 2039 go support for non latin documents when fetching supported documents

### PR DESCRIPTION
**Pull request by mehmet-yoti**
## IDV

**Added**

- Support for non latin documents when fetching supported documents

**Requested with**
```go	
result, err := client.GetSupportedDocuments(true)
```

### CODE REVIEW

**Review by fofiuiancu — CHANGES_REQUESTED** [review]
it seems that his PR contains changes for multiple PRs?

**Comment by mehmet-yoti** [comment]
Yes it has dependencies with multiple commits.

File: docscan/client.go [hunk 1]
```diff
@@ -243,12 +243,17 @@ func (c *Client) DeleteMediaContent(sessionID, mediaID string) error {
 }
 
 // GetSupportedDocuments gets a slice of supported documents
-func (c *Client) GetSupportedDocuments() (*supported.DocumentsResponse, error) {
+func (c *Client) GetSupportedDocuments(isStrictlyLatin bool) (*supported.DocumentsResponse, error) {
```
Comment (fofiuiancu) [inline docscan/client.go:246]:
```suggestion
func (c *Client) GetSupportedDocuments(includeNonLatin bool) (*supported.DocumentsResponse, error) {
```

File: docscan/client.go [hunk 2]
```diff
@@ -243,12 +243,17 @@ func (c *Client) DeleteMediaContent(sessionID, mediaID string) error {
...
+	includeNonLatin := "0"
```
Comment (fofiuiancu) [inline docscan/client.go:247]: this logic is the inverse of what we need and can be removed by having includeNonLatin as the function's param

File: docscan/client.go [hunk 3]
```diff
@@ -243,12 +243,17 @@ func (c *Client) DeleteMediaContent(sessionID, mediaID string) error {
...
+	if isStrictlyLatin {
+		includeNonLatin = "1"
+	}
 	request, err := (&requests.SignedRequest{
 		Key:        c.Key,
 		HTTPMethod: http.MethodGet,
 		BaseURL:    c.apiURL,
 		Endpoint:   getSupportedDocumentsPath(),
+		Params:     map[string]string{"includeNonLatin": includeNonLatin},
```
Comment (fofiuiancu) [inline docscan/client.go:256]:
```suggestion
		Params:     map[string]string{"includeNonLatin": strconv.FormatBool(includeNonLatin)},
```

**Review by fofiuiancu — CHANGES_REQUESTED** [review]

File: docscan/client.go [hunk 4]
```diff
@@ -243,12 +244,14 @@ func (c *Client) DeleteMediaContent(sessionID, mediaID string) error {
 }
 
 // GetSupportedDocuments gets a slice of supported documents
-func (c *Client) GetSupportedDocuments() (*supported.DocumentsResponse, error) {
+func (c *Client) GetSupportedDocuments(includeNonLatin bool) (*supported.DocumentsResponse, error) {
```
Comment (fofiuiancu) [inline docscan/client.go:247]: might be a breaking change

Comment (mehmet-yoti) [reply docscan/client.go]: Preserved GetSupportedDocuments to maintain backward compatibility added different method for GetSupportedDocumentsWithNonLatin

**Review by fofiuiancu — APPROVED** [review]

**Pull request merged**